ecosystem/sep-0010.md

```diff
@@ -139,7 +139,75 @@ operation of the challenge is the **Home Domain**, and that the web auth domain
+
+Authorized `WEB_AUTH_ENDPOINT` accepts `Authorization Bearer: <token>`, where token is [JSON Web Token](https://jwt.io/)
+authorizing the request.
+
+Client must specify the following claims:
+
+- `iat` (the time at which the JWT was issued
+  [RFC7519, Section 4.1.6](https://tools.ietf.org/html/rfc7519#section-4.1.6)) — current timestamp (`1530644093`)
+- `exp` (the expiration time on or after which the JWT must not be accepted for processing,
+  [RFC7519, Section 4.1.4](https://tools.ietf.org/html/rfc7519#section-4.1.4)) — a server can pick its own expiration
+  period for the token (`1530730493`)
+- `url` full URL of the request, including scheme, path and query parameters.
+
+Client must then correctly sign the payload with appropriate Stellar private key. To choose the private key client
+application should follow this steps:
+
+- If `client_domain` is specified, the token must be signed with the `SIGNING_KEY`. (See
+  [Verifying the Client Domain](#verifying-the-client-domain) section for more information)
+- Otherwise, the token must be signed with a private key of the **Client Account** passed in the `account` field.
+
+Token then should be signed with the selected private key, using **ed25519** algorithm. An appropriate header must be
+included
+
+```json
+{
+  "alg": "EdDSA"
+}
+```
+
+Example of the valid signed token in base64 format:
+
+```
+eyJhbGciOiJFZERTQSJ9.eyJqdGkiOiIzYjExYTRhYS1lMmE4LTRkYTUtYTUxOC1iMDU5YWMwMjUwOGYiLCJpYXQiOjE3MTA5NzY2MDcsImV4cCI6MTAzNTA5NzY2MDcsInVybCI6Imh0dHBzOi8vYXV0aC5leGFtcGxlLmNvbS8_YWNjb3VudD1HQ0lCVUNHUE9IV01NTUZQRlREV0JTVkhRUlQ0RElCSjdBRDZCWkpZRElUQksyTENWQllXN0hVUSZjbGllbnRfZG9tYWluPWV4YW1wbGUtd2FsbGV0LnN0ZWxsYXIub3JnIn0.FhX1sxGZcFMEwfRE4W3fe8uB5PUtBBS3_nkKUVHx9Ulw3Kc3OvJ5bwLRXuWRpBndvuK2S-yKdKBUCLqd4QO-Cg
+```
+
+This token was signed with the `GCADAIACE6CRWGOB3HJRXIOHMQEUKPNLAUIYFIE26F3OK72RSHGAMARK` key, belonging to a
+hypothetical domain `example-wallet.stellar.org`
+
+Another example of the valid signed token:
+
+```
+eyJhbGciOiJFZERTQSJ9.eyJqdGkiOiI1YWExYTNhYi0zY2Q2LTRkYWYtODQyMi0xYmI5Zjc1NTI1YjEiLCJpYXQiOjE3MTA5NzY5MDYsImV4cCI6MTAzNTA5NzY5MDYsInVybCI6Imh0dHBzOi8vYXV0aC5leGFtcGxlLmNvbS8_YWNjb3VudD1HQ1hYSDZBWUpVVlRER0lIVDQyT1pOTUYzTEhDVjRET0tDWDZISERLV0VDVVpZWERaU1daTjZIUyZtZW1vPTEyMzQ1NjcifQ.Gi7aPIwLBss54DJzKhi9lRlUViCkLbbjT0SQFrG9eGGt7TAGD1VeLfj2LFBCOoCFytXWDufplY4iJk4XG_jwDw
+```
+
+This time the token is signed with a **Client Account** () private key
+
+##### Verifying request header
+
+When server receives a request with an `Authorization` header, server should validate the token before processing with
+issuing a challenge transaction.  
+First, server must use the same steps of choosing the public key (see above), and verify **ed25519** signature of the
+token. Next, server must verify that:
+
+1. URL's path must match `<WEB_AUTH_ENDPOINT>`
+2. URL's query parameters must match request's parameters. Server should address possible permutations of the query
+   parameters.
```

Comment:
What do you mean by "address possible permutations of the query parameters"?

Reply:
If the server receives request
```
https://auth.example.com/?account=GCXXH6AYJUVTDGIHT42OZNMF3LHCV4DOKCX6HHDKWECUZYXDZSWZN6HS&memo=1234567
``` 
But in JWT url is
```
https://auth.example.com/?memo=1234567&account=GCXXH6AYJUVTDGIHT42OZNMF3LHCV4DOKCX6HHDKWECUZYXDZSWZN6HS
```
It should not be rejected (not sure how to word it better)

Reply:
I see, maybe we can avoid this problem by moving the query parameters to actual JWT fields?

So for example, instead of having the server validate the query parameters of the URL, the client can put the account, memo, and client_domain in fields within the JWT?

```js
{
  "iat": "...",
  "exp": "...",
  "url": "...",
  "sub": "G...:12314213",
  "client_domain": "..."
}
```

Reply:
Or maybe if we're moving parameters outside the URL, se can use `host` instead of `url`?

Reply:
I think we can do that. We'll have query parameters, `WEB_AUTH_ENDPOINT` and `host` as part of JWT. Then anchor will validate that query params, endpoint and host are matching